source/Missions/Services/Network/NetworkAgentRegistry.cs

```diff
@@ -80,18 +109,50 @@ public bool RegisterNetworkControlledAgent(NetPeer peer, Guid agentId, Agent age
         {
             if (m_OtherAgents.TryGetValue(peer, out AgentGroupController controller))
             {
-                if (controller.Contains(agent)) return false;
-                if (controller.Contains(agentId)) return false;
+                if (controller.Contains(agent) || controller.Contains(agentId)) return false;
+
+                controller.AddAgent(agentId, agent);
+                m_AgentToId.Add(agent, agentId);
+            }
+            else
+            {
+                AgentGroupController newGroupController = new AgentGroupController();
+                newGroupController.AddAgent(agentId, agent);
+                m_AgentToId.Add(agent, agentId);
+                m_OtherAgents.Add(peer, newGroupController);
+            }
+
+            return true;
+        }
+
+        public bool RegisterPlayerAgent(Guid agentId, Agent agent)
+        {
+            if (m_AgentToId.ContainsKey(agent) ||
+                m_PlayerAgents.ContainsKey(agentId)) return false;
+
+            m_PlayerAgents.Add(agentId, agent);
+            m_AgentToId.Add(agent, agentId);
+
+            return true;
+        }
+
+        public bool RegisterNetworkPlayerAgent(NetPeer peer, Guid agentId, Agent agent)
+        {
+            if (m_OtherAgents.TryGetValue(peer, out AgentGroupController controller))
+            {
+                if (controller.Contains(agent) || controller.Contains(agentId)) return false;
 
                 controller.AddAgent(agentId, agent);
                 m_AgentToId.Add(agent, agentId);
+                m_PlayerAgents.Add(agentId, agent);
             }
             else
             {
                 AgentGroupController newGroupController = new AgentGroupController();
                 newGroupController.AddAgent(agentId, agent);
                 m_AgentToId.Add(agent, agentId);
                 m_OtherAgents.Add(peer, newGroupController);
+                m_PlayerAgents.Add(agentId, agent);
```

Comment:
```suggestion
```

Reply:
that would make the whole PlayerAgents dictionary obsolete, wouldn't it?